### PR TITLE
obs-ffmpeg: Fix broken mpegts output

### PR DESCRIPTION
**Pull request by pkviet**
### Description
This recent commit [1] broke srt & rist streaming because obs_output_video now returns a null ptr for encoded outputs. This fixes it by retrieving the pointer from the encoder.

[1] https://github.com/obsproject/obs-studio/commit/fb57eff212fbe5e777cf559288bc70bf67d9c428

### Motivation and Context
Fix streaming with SRT / RIST.

### How Has This Been Tested?
SRT streaming works again.

### Types of changes
- Bug fix (non-breaking change which fixes an issue)

### Checklist:
- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

File: plugins/obs-ffmpeg/obs-ffmpeg-mpegts.c [hunk 1]
```diff
@@ -885,8 +885,22 @@ static bool set_config(struct ffmpeg_output *stream)
...
+	config.scale_height = config.height;
+
+	// 2.b) set video codec & id from video encoder
+	obs_encoder_t *vencoder = obs_output_get_video_encoder(stream->output);
+	config.video_encoder = obs_encoder_get_codec(vencoder);
+	if (strcmp(config.video_encoder, "h264") == 0)
+		config.video_encoder_id = AV_CODEC_ID_H264;
+	else
+		config.video_encoder_id = AV_CODEC_ID_AV1;
+	// 2.c) set video format from obs to FFmpeg
```
Comment (RytoEX) [inline plugins/obs-ffmpeg/obs-ffmpeg-mpegts.c:902]: Nit: please always leave a blank line after an unenclosed `if`, `if-else`, or `if-else if-else` statement to make it clear that we have exited that statement.

Comment (RytoEX) [inline plugins/obs-ffmpeg/obs-ffmpeg-mpegts.c:902]:
```suggestion
	// 2.c) set video format from OBS to FFmpeg
```

File: plugins/obs-ffmpeg/obs-ffmpeg-mpegts.c [hunk 2]
```diff
@@ -885,8 +885,22 @@ static bool set_config(struct ffmpeg_output *stream)
...
-	// 2.a) set video format from obs to FFmpeg
-	video_t *video = obs_output_video(stream->output);
+
+	// 2.a) set width & height
+	config.width = (int)obs_output_get_width(stream->output);
+	config.height = (int)obs_output_get_height(stream->output);
+	config.scale_width = config.width;
+	config.scale_height = config.height;
+
+	// 2.b) set video codec & id from video encoder
```
Comment (RytoEX) [inline plugins/obs-ffmpeg/obs-ffmpeg-mpegts.c:895]:
```suggestion
	// 2.b) set video codec & ID from video encoder
```

**Review by RytoEX — CHANGES_REQUESTED** [review]

**Review by RytoEX — CHANGES_REQUESTED** [review]
Commit message nits:
* srt -> SRT
* rist -> RIST

File: plugins/obs-ffmpeg/obs-ffmpeg-mpegts.c (hunk 1 above)
Comment (pkviet) [reply plugins/obs-ffmpeg/obs-ffmpeg-mpegts.c:902]: done

Comment (pkviet) [reply plugins/obs-ffmpeg/obs-ffmpeg-mpegts.c:902]: done

**Comment by pkviet** [comment]
> Commit message nits:
> 
> * srt -> SRT
> * rist -> RIST

fixed

**Comment by norihiro** [comment]
The modification looks good to me, checked by eyeball and by running it.
(Reordering the steps `set width & height` and `set video codec & id from video encoder` is not necessary for bugfix but should not cause any problem.)

**Review by RytoEX — APPROVED** [review]
Seems fine. @tt2468 approved off-thread.

**Pull request merged**